packages/typespec-client-generator-core/test/types.test.ts

```diff
@@ -1450,17 +1452,40 @@ describe("typespec-client-generator-core: types", () => {
 
       const catValue = values.find((x) => x.name === "Cat")!;
       strictEqual(catValue.value, "cat");
+      strictEqual(catValue.description, "Cat");
       strictEqual(catValue.enumType, petKind);
       strictEqual(catValue.valueType, petKind.valueType);
       strictEqual(catValue.kind, "enumvalue");
 
       const dogValue = values.find((x) => x.name === "Dog")!;
       strictEqual(dogValue.value, "dog");
+      strictEqual(dogValue.description, "Dog");
       strictEqual(dogValue.enumType, petKind);
       strictEqual(dogValue.valueType, petKind.valueType);
       strictEqual(dogValue.kind, "enumvalue");
     });
 
+    it("property of anonymous union as enum", async () => {
+      await runner.compileWithBuiltInService(`
+      model Pet {
```

Comment:
we don't want to create an enum from an anonymous union, we only do it if the union is named. let me roll back this part of the pr, rest looks good as far as i can see

Reply:
is this not true anymore?

Reply:
i didn't know we changed this, i know @lmazuel before really didn't want to generate a named enum from anonymous unions. can you approve it as is and @tadelesh depending on what @lmazuel says can you make a separate PR if we do want to name anonymous unions?